docs/content/3.middleware/2.guides/3.extensions.md

```diff
@@ -117,3 +121,104 @@ curl '{SERVER_DOMAIN}/api/magento/getProductBySku`' \
   -H 'content-type: application/json' \
   -d '[{ "id": 1, "name": "John" }]'
 ```
+
+### Creating a configurable extension
+
+In more complex cases, you might want to create an extension that can be configured.
+
+Imagine that you're preparing an extension that would add cache control headers to responses based on the configuration.
+It might look like this:
+
+```ts
```

Comment:
I would recommend either removing the code snippet of the "non-recommended" way or having a comment at the top that says that this is the less reusable way/suboptimal to do this.

That way someone doesn't start reading this code snippet thinking it's what we recommend and copy/pasting it into their project before finishing the doc

Reply:
@mattmaribojoc suggestion applied!